_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,194 @@
+---
+title: "Stay on topic: Using Kafka to trigger your OpenFaaS functions"
+description: Martin outlines how you can use Kafka and subscribe OpenFaaS function to a topic using kafka-connector plugin.
+date: 2019-03-14
+image: /images/google-form-response/dots.jpg
+categories:
+  - kafka
+  - tutorial
+  - examples
+author_staff_member: martin
+dark_background: true
+---
+
+In this post I will show you how to subscribe your OpenFaaS functions to Kafka topics when running your functions via Kubernetes.
+
+OpenFaaS is production ready Serverless framework, and often finds itself as a part of bigger eco-system of software where the disparate parts may communicate with each-other in different ways. The OpenFaaS gateway exposes REST endpoints, but what if our services don't use HTTP?
+
+Kafka is stream-processing platform and at a high level the important concepts are Producers, Consumers and Brokers which communicate with each-other via massages attached to Topics. The way it works is that the Producer is sending messages on specific Topics to the Broker, then the Consumers poll all the topics of interest from the Brokers. The approach is fitting for distributed systems, because it decouples the communication between your services.
+
+You can find out more by heading to the [Apache Kafka Documentation](https://kafka.apache.org/documentation/).
+
+## Kafka Connector
+
+Kafka Connector plugin is the solution to your problem when it comes to integrating OpenFaaS as a component of a bigger distributed system which is using Kafka to make communication between its parts. You just need to apply annotation with the topic you would like to subscribe your function to and then run the connector with your Kafka configuration.
+
+Here is a diagram which shows how the component works:
+
+![Form Options](/images/kafka-connector/kafka-connector-detailed-overview.png)
```

Comment:
What is `Form Options`?

Reply:
Adds the picture, at least that's how other blogs do it took it from there

Reply:
changed to Kafka Diagram